conda/cli/main_export.py

```diff
@@ -96,38 +126,77 @@ def configure_parser(sub_parsers: _SubParsersAction, **kwargs) -> ArgumentParser
 
 # TODO Make this aware of channels that were used to install packages
 def execute(args: Namespace, parser: ArgumentParser) -> int:
-    from ..base.context import context, determine_target_prefix, env_name
-    from ..env.env import from_environment
+    from ..base.context import env_name
     from .common import stdout_json
 
-    prefix = determine_target_prefix(context, args)
-    env = from_environment(
-        env_name(prefix),
-        prefix,
+    # Validate export format early before doing expensive environment operations
+    available_formats = list(_get_available_export_formats())
+
+    # Early format validation - fail fast if format is unsupported
+    target_format = args.format
+    environment_exporter = None
+
+    # If explicit format provided, use it and find the appropriate exporter
+    if target_format is not NULL:
+        environment_exporter = (
+            context.plugin_manager.get_environment_exporter_by_format(target_format)
+        )
+        if not environment_exporter:
+            raise CondaValueError(
+                f"Unknown export format '{target_format}'. "
+                f"Available formats: {', '.join(available_formats)}."
```

Comment:
I've found the `dashlist` to be much better for this

```suggestion
                f"Available formats: {dashlist(available_formats)}."
```